src/core/experimental/refcount.d

```diff
@@ -0,0 +1,545 @@
+module core.experimental.refcount;
```

Comment:
Please sufficiently document this module.  It very much helps with the review.

Reply:
Would be nice with a usage example.